app/firebase.server.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  cert,
+  initializeApp,
+  getApps,
+  getApp,
+  App,
+  ServiceAccount,
+} from "firebase-admin/app";
+import { getAuth, Auth } from "firebase-admin/auth";
```

Comment:
On the server, we rely on the `firebase-admin` package to verify the tokens sent by the frontend. Here, none of the credentials are exposed to contributors. If someone wants to run auth locally, I've provided an emulator that "verifies" tokens without making actual calls to a service.